docs/migration.md

```diff
@@ -4,7 +4,7 @@
 
 ### Dependency Changes
 
-Python 3.7 support has been dropped. Please upgrade to Python 3.8 or later to
+Python 3.7 support has been dropped. Please upgrade to Python 3.9 or later to
```

Comment:
@eliotwrobson Why did this change? This seemed logically correct before (i.e. if you drop 3.7 support, then 3.8 is the minimum supported version, not 3.9).

Reply:
We just waited too long to update this: https://devguide.python.org/versions/

Reply:
@eliotwrobson Yes, but even if 3.8 hits EOL, dropping support for it still necessitates a semver-major version bump because you it technically constitutes a breaking change for users still on Python 3.8 for whatever reason.

Not that I would be opposed to dropping 3.8 support now that it _is_ EOL, but I would prefer that the release version accurately reflects the drop in support. With that, I might suggest we hold off on this PR until we release what has already been merged (i.e. to make for the nice 8.4.1 release you mentioned), then release v9 with drop for 3.8 support.

Thoughts?

Reply:
I think that makes sense, though doing multiple consecutive releases seems a little overkill for something like this. I'd rather just release a minimal 9.0 with everything we have so far (I think develop only has one additional feature that is just expanded unicode support and a bugfix).

Reply:
@eliotwrobson I'm inclined to agree, actually. Let's just stick to a single v9.0.0 release.

As far as updating the migration guide, I think I'd still recommend leaving at 3.8, though. Because otherwise, are we setting a precedent where we feel obligated to keep track of core Python's EOL timeline and update all past Python-version-dropping major releases in the migration guide?

Reply:
I agree, I'll go ahead and add a new v9 section 👍🏽

Reply:
Added!